s3repo/model.py

```diff
@@ -0,0 +1,200 @@
+
+                mkrepo_cmd = [
+                    'python3',
+                    './third_party/mkrepo/mkrepo.py',
+                    '--s3-access-key-id',
+                    str(self.s3_settings['access_key_id']),
+                    '--s3-secret-access-key',
+                    str(self.s3_settings['secret_access_key']),
+                    '--s3-endpoint',
+                    str(self.s3_settings['endpoint_url']),
+                    '--s3-region',
+                    str(self.s3_settings['region']),
+                    's3://' + self.s3_settings['bucket_name'] + '/' + sync_repo
+                ]
+                with sp.Popen(mkrepo_cmd) as mkrepo_ps:
+                    result = mkrepo_ps.wait()
+                    if result != 0:
+                        self.sync_lock.acquire()
+                        self.unsync_repos.add(sync_repo)
+                        self.sync_lock.release()
+            else:
+                # The "unsync_repos" set is empty.
+                # Let's just wait a while.
+                time.sleep(5)
+
+    def put_package(self, package):
+        """Load the package to S3."""
+        repos_to_load = []
+        if package.repo == 'all':
+            repos_to_load = self.get_supported_distrs()['supported_versions']
+        else:
+            repos_to_load.append(package.repo)
+
+        origin_files_on_s3 = {}
+        for repo in repos_to_load:
+            base_path = '/'.join([
+                self.s3_settings['base_path'],
+                package.repos_type,
+                repo,
+                package.dist
+            ])
+
+            unsync_repos_local = set()
+            for filename, file in package.files.items():
+                path = ''
+                repo_path = ''
+                if re.fullmatch(r'.*\.(x86_64|noarch)\.rpm', filename):
+                    repo_path = '/'.join([
+                        base_path,package.version,
+                        'x86_64'
+                    ])
+                    path = '/'.join([
+                        repo_path,
+                        'Packages',
+                        filename
+                    ])
+                elif re.fullmatch(r'.*\.src\.rpm', filename):
+                    repo_path = '/'.join([
+                        base_path,
+                        package.version,
+                        'SRPMS'
+                    ])
+                    path = '/'.join([repo_path, 'Packages', filename])
+                elif re.fullmatch(r'.*\.(deb|dsc|tar\.xz|tar\.gz)', filename):
+                    repo_path = base_path
+                    path = '/'.join([
+                        base_path,
+                        'pool',
+                        package.version,
+                        'main',
+                        filename[:1],
+                        filename.partition('_')[0],
+                        filename
+                    ])
+                else:
+                    continue
```

Comment:
> When a caller (view's) filename check is incorrect (is not consistent with this code)?

Yep.

> So we should raise an error here rather than ignore the file.

Added raising an error.